spec/service_models/miq_ae_service_manageiq-providers-openstack-cloud_manager-cloud_volume_spec.rb

```diff
@@ -1,26 +1,26 @@
 describe MiqAeMethodService::MiqAeServiceUser do
-  let(:cloud_volume)         { FactoryBot.create(:cloud_volume_openstack) }
+  let(:cloud_volume)         do
+    FactoryBot.create(:cloud_volume_openstack, :ext_management_system => FactoryBot.create(:ems_openstack))
+  end
   let(:service_cloud_volume) do
     MiqAeMethodService::MiqAeServiceManageIQ_Providers_Openstack_CloudManager_CloudVolume.find(cloud_volume.id)
   end
 
   before do
-    zone = FactoryBot.create(:zone)
-    allow_any_instance_of(CloudVolume).to receive(:my_zone).and_return(zone.name)
-    allow(MiqServer).to receive(:my_zone).and_return(zone.name)
     @base_queue_options = {
       :class_name  => cloud_volume.class.name,
       :instance_id => cloud_volume.id,
-      :zone        => zone.name,
+      :zone        => cloud_volume.ext_management_system.my_zone,
       :role        => 'ems_operations',
+      :queue_name  => cloud_volume.queue_name_for_ems_operations,
       :task_id     => nil
     }
   end
 
   it "#backup_create async" do
     service_cloud_volume.backup_create('test backup', false)
 
-    expect(MiqQueue.first).to have_attributes(
+    expect(MiqQueue.last).to have_attributes(
```

Comment:
When you create an OpenStack EMS with FactoryBot it queues a couple of queue messages to stop the event monitor for the child classes so the queue item we care about is actually the last one.

```
(byebug) MiqQueue.pluck(:class_name, :method_name)
[["ManageIQ::Providers::Openstack::CloudManager", "stop_event_monitor"], ["ManageIQ::Providers::Openstack::NetworkManager", "stop_event_monitor"], ["ManageIQ::Providers::Openstack::StorageManager::CinderManager", "stop_event_monitor"], ["ManageIQ::Providers::Openstack::CloudManager::CloudVolume", "backup_create"]]
```